lib/game_session.rb

```diff
@@ -31,11 +31,13 @@ def run_game_loop
   end
 
   def display_current_board_status
-    if @rules.winner?(@board.grid)
+    full_board = @board.full_board?(@board.grid)
+
+    if @outcome.winner?(@board.get_all_move_combos)
       @display1.print_winner(@current_player)
-    elsif @rules.is_tie?(@board.grid)
+    elsif @outcome.is_tie?(@board.get_all_move_combos, full_board)
       @display1.print_tie
-    else !@rules.winner?(@board.grid) || !@rules.is_tie?(@board.grid)
+    else !@outcome.winner?(@board.get_all_move_combos) || !@outcome.is_tie?(@board.get_all_move_combos)
```

Comment:
Do you have to check for winning game or tie again in your `else` branch? It can just be `else`!

Reply:
Good call! I don't have to check again in the else branch :)